src/Cake.Npm/AddUser/NpmAddUserSettings.cs

```diff
@@ -0,0 +1,96 @@
+﻿namespace Cake.Npm.AddUser
+{
+    using System;
+    using System.Linq;
+    using Core;
+    using Core.IO;
+
+    /// <summary>
+    /// Contains settings used by <see cref="NpmAddUser"/>.
+    /// </summary>
+    public class NpmAddUserSettings : NpmSettings
+    {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="NpmAddUserSettings"/> class.
+        /// </summary>
+        public NpmAddUserSettings()
+            : base("adduser")
+        {
+        }
+
+        /// <summary>
+        /// Gets or sets a value indicating which registry we should point to. Defaulted to whatever the NPM configuration is.
+        /// </summary>
+        /// <value>
+        /// Defaults to NPM registry.
+        /// </value>
+        public Uri Registry { get; set; }
+
+        /// <summary>
+        /// Gets or sets a value indicating the specificed scope for the user and login credentials provided. Defaults to <c></c>
+        /// </summary>
+        /// <value>
+        /// Defaults to none.
+        /// </value>
+        public string Scope { get; set; }
+
+        /// <summary>
+        /// Gets a value indicating that all requests to the given registry should 
+        /// include authorization information. 
+        /// </summary>
+        /// <value>
+        /// <c>true</c> to always provided auth, else <c>false</c>. Defaults to <c>false</c>.
+        /// </value>
+        public bool AlwaysAuth { get; set; }
+
+        /// <summary>
+        /// Gets or sets the authentication stragety to use.
+        /// </summary>
+        public AuthType AuthType { get; set; }
+
+        /// <summary>
+        /// Evaluates the settings and writes them to <paramref name="args"/>.
+        /// </summary>
+        /// <param name="args">The argument builder into which the settings should be written.</param>
+        protected override void EvaluateCore(ProcessArgumentBuilder args)
+        {
+            base.EvaluateCore(args);
+
+            if (Registry != null)
+            {
+                args.AppendSwitch("--registry", Registry.ToString());
+            }
+
+            if (!string.IsNullOrWhiteSpace(Scope))
+            {
+                if (!Scope.StartsWith("@"))
+                {
+                    throw new ArgumentException("Scope should start with @", nameof(Scope));
+                }
+                args.AppendSwitch("--scope", Scope.ToString());
+            }
+
+            if (AlwaysAuth)
+            {
+                args.Append("--always-auth");
+            }
+
+            var authType = "legacy";
```

Comment:
IMHO code would be clearer and more stable if all possible cases are handled in the `switch` statement and if an unknown value is entered an exception is thrown (instead of silently falling back to legacy)